added support for VCL

himyouten opened the pull request.

himyouten commented:
I haven't done any tests...

fstehle commented:
Thanks a lot for your contribution.
The code looks great so far.
Can you please add a test (`tests/test_fastly_vcl.py`) for it (look at https://github.com/Jimdo/ansible-fastly/blob/540bdb0620c12319133c396eeae137ab702480fa/tests/test_fastly_gzip.py for an example)?
Can you please enhance the README for the `vcls` module options for it as well?

himyouten commented:
I've added S3 logging support and updated the README, but I can't figure out the tests, sorry.  If someone else wants to complete the tests that'd be grand!

himyouten commented:
I figured out the tests!  Wrote tests for all new functions - and added a paragraph for how to do the tests for next time I might need it :-)

xiongchiamiov commented:
It would be lovely if the vcl could be provided from a file instead of only an inline string; that would help keep things tidy when dealing with lots of vcl.  But perhaps that's something out of the scope of this pull.

himyouten commented:
Hi - sorry - been a while - but I recently just added dictionary support - which required doing a clone instead of a full rebuild.  Unfortunately, I don't think I'll get around to doing the tests this time.  Its a pretty major rework.

himyouten commented:
@xiongchiamiov I'd love to be able to do that too!  But I haven't had the time to figure out how to use Ansible's templating calls.  I currently load the files using vars and lookup('template')
